rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java

```diff
@@ -141,7 +141,33 @@ public String getResourceGroupId() {
     @Override
     public String getResourceId() {
         if (jdbcUrl.contains("?")) {
-            return jdbcUrl.substring(0, jdbcUrl.indexOf('?'));
+            StringBuilder jdbcUrlBuilder = new StringBuilder();
```

Comment:
According to dbType, non-pg remains the same

Reply:
fixed